### PR TITLE
Core/Movement: Allow using walk when chasing v2

### PR DESCRIPTION
**Pull request by Sorikoff**
**Changes proposed:**
Allow setting chase walk state for each creature through DB table `creature_template_movement`/`creature_movement_override`. Values for new column `Chase` are following:

0. run always;
1. respect script settings;
2. walk always.

**Target branch(es):** 3.3.5

**Tests performed:** Does build, _works-in-game_ (not everything tested)

### CODE REVIEW

File: src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.cpp [hunk 1]
```diff
@@ -203,9 +201,18 @@ bool ChaseMovementGenerator::Update(Unit* owner, uint32 diff)
 
             owner->AddUnitState(UNIT_STATE_CHASE_MOVE);
 
+            bool walk = false;
+            if (cOwner)
+            {
+                if (cOwner->GetMovementTemplate().CanChaseWalking() && owner->IsWalking())
```
Comment (Shauren) [inline src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.cpp]: switch on Chase field would look nicer imo
You need to ignore it for pets (any player controlled? needs verification)

File: src/server/game/Entities/Creature/CreatureData.h [hunk 2]
```diff
@@ -93,20 +93,35 @@ enum class CreatureFlightMovementType : uint8
...
+
+    Max
+};
+
 struct TC_GAME_API CreatureMovementData
 {
-    CreatureMovementData() : Ground(CreatureGroundMovementType::Run), Flight(CreatureFlightMovementType::None), Swim(true), Rooted(false) { }
+    CreatureMovementData() : Ground(CreatureGroundMovementType::Run), Flight(CreatureFlightMovementType::None),
+                             Swim(true), Rooted(false), Chase(CreatureChaseMovementType::Run)
+    {}
```
Comment (ccrs) [inline src/server/game/Entities/Creature/CreatureData.h]: image

File: src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.cpp (hunk 1 above)
Comment (Sorikoff) [reply src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.cpp]: Agree on switch. Ignores pets now.

File: src/server/game/Entities/Creature/CreatureData.h (hunk 2 above)
Comment (Sorikoff) [reply src/server/game/Entities/Creature/CreatureData.h]: Whoops.

**Comment by jackpoz** [comment]
Can this PR be merged or is there anything left to do ?

**Comment by Sorikoff** [comment]
Looks good to me.

**Comment by illfated** [comment]
Looks like progress to me, although I am not a real coder.

**Comment by Kittnz** [comment]
Why is this even needed?

**Comment by Sorikoff** [comment]
Some creatures must walk when chasing. This pull request allows to manage this behaviour through DB.

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Comment by illfated** [comment]
Yes, thank you for adding to the source content. 👍

**Comment by jackpoz** [comment]
This PR didn't update WORLD_SEL_CREATURE_TEMPLATE , causing ".reload creature_template <entry>" to crash the server

**Pull request merged**